[Term Entry] C++ Math Functions: asinh() (#1074)
Codecademy/docs: content/cpp/concepts/math-functions/terms/asinh/asinh.md

---
Title: 'asinh'

> Description: 'Returns the inverse hyperbolic sine of argument in radian.'

Reviewer: ```suggestion
Description: 'Returns the inverse hyperbolic sine of an argument in radians.'
```

Subjects:
  - 'Computer Science'
Tags:
  - 'Functions'
  - 'Arithmetic'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

> Returns the inverse hyperbolic sine of an angle.

Reviewer: ```suggestion
Returns the inverse hyperbolic sine of an angle given in radians.
```

## Syntax

```cpp
asinh(n)
```

> Please include the `cmath` header file to use this function.

Reviewer: ```suggestion
The `cmath` header file must be included to use this function.
```

## Example

Use `asinh()` to return the inverse hyperbolic sine of `3.14159`:

```cpp
#include <iostream>
#include <cmath>

int main() {
    double x = 3.14159;
    double result;
    result = std::asinh(x);
    std::cout << result << " radian" << "\n";

> ```

Reviewer: Include a ```shell block with the output of this example.

Reviewer: @sudojarvis, This still needs to be addressed

## Codebyte Example

> 

Reviewer: Have a sentence introducing the codebyte example.

---
Title: 'asinh'

> Description: 'Returns the inverse hyperbolic sine of an argument in radians.'

Reviewer: ```suggestion
Description: 'Returns the inverse hyperbolic sine of an angle given in radians.'
```

Subjects:
  - 'Computer Science'
Tags:
  - 'Functions'
  - 'Arithmetic'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

Returns the inverse hyperbolic sine of an angle given in radians.

## Syntax

```cpp
asinh(n)
```

The `cmath` header file must be included to use this function.

## Example

Use `asinh()` to return the inverse hyperbolic sine of `3.14159`:

```shell
#include <iostream>
#include <cmath>

int main() {
    double x = 3.14159;
    double result;
    result = std::asinh(x);
    std::cout << result << " radian" << "\n";
```

## Codebyte Example

> The `asinh()` function take argument as `4.55` and returns the inverse hyperbolic sine of `4.55`:

Reviewer: ```suggestion
The `asinh()` function takes argument as `4.55` and returns the inverse hyperbolic sine:
```

---
Title: 'asinh'
Description: 'Returns the inverse hyperbolic sine of an angle given in radians.'
Subjects:
  - 'Computer Science'
Tags:
  - 'Functions'
  - 'Arithmetic'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

Returns the inverse hyperbolic sine of an angle given in radians.

## Syntax

```cpp
asinh(n)
```

The `cmath` header file must be included to use this function.

## Example

> Use `asinh()` to return the inverse hyperbolic sine of `3.14159`:

Reviewer: ```suggestion
This example uses the `asinh()` function to return the inverse hyperbolic sine of `3.14159`:
```

> ```shell

Reviewer 2: C++ code should be a C++ code block, not a shell. Use

\```cpp
\```

#include <iostream>
#include <cmath>

int main() {
    double x = 3.14159;
    double result;
    result = std::asinh(x);
    std::cout << result << " radian" << "\n";
```

> ## Output

Reviewer: ```suggestion
This example results in the following output:
```

```shell
1.8623 radian
```

## Codebyte Example

> The `asinh()` function takes argument as `4.55` and returns the inverse hyperbolic sine:

Reviewer: ```suggestion
This example uses the `asinh()` function to take an  argument of `4.55` and return the inverse hyperbolic sine:
```

---

> Title: 'asinh'

Reviewer 3: ```suggestion
Title: 'asinh()'
```

Reviewer 3: If the entry is about a method/function, the `Title` should include any punctuation such as parentheses and/or periods. 😄

Description: 'Returns the inverse hyperbolic sine of an angle given in radians.'
Subjects:
  - 'Computer Science'
Tags:
  - 'Functions'
  - 'Arithmetic'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

> Returns the inverse hyperbolic sine of an angle given in radians.

Reviewer 3: ```suggestion
The **`asinh()`** function returns the inverse hyperbolic sine of an angle given in radians.
```

## Syntax

> ```cpp

Reviewer 3: ```suggestion
```pseudo
```

asinh(n)
```

> The `cmath` header file must be included to use this function.

Reviewer 3: ```suggestion
The `n` parameter can be a floating-point value (e.g., a `long` or `double`) or an integral type (e.g., an `int` or `bool`).
```

I think that mentioning the `cmath` header isn't necessary here since this point is stated in the parent "Math Functions" entry. Including info about the parameter(s) is more relevant to this section.

## Example

This example uses the `asinh()` function to return the inverse hyperbolic sine of `3.14159`:

```cpp
#include <iostream>
#include <cmath>

> int main() {
>     double x = 3.14159;
>     double result;
>     result = std::asinh(x);
>     std::cout << result << " radian" << "\n";
>     }

Reviewer 3: ```suggestion
int main() {
  double x = 3.14159;
  double result;
  result = std::asinh(x);
  std::cout << result << " radian" << "\n";
}
```

```

This example results in the following output:

```shell
1.8623 radian
```

## Codebyte Example

> This example uses the `asinh()` function to take an  argument of `4.55` and return the inverse hyperbolic sine:

Reviewer 3: ```suggestion
This example uses the `asinh()` function to take an argument of `4.55` and return the inverse hyperbolic sine:
```

```codebyte/cpp
#include <iostream>
#include <cmath>

> int main() {
>     double x = 4.55;
>     double result;
>     result = std::asinh(x);
>     std::cout << result << " radian" << "\n";
>     // OUTPUT: 2.220137 radian
> }

Reviewer 3: ```suggestion
int main() {
  double x = 4.55;
  double result;
  result = std::asinh(x);
  std::cout << result << " radian" << "\n";
}
```

Reviewer 3: No need to have output comments/snippets in the "Codebyte Example" 😄
